### DISCUSSION TITLE
Self Hosted Registry Help Needed

### DISCUSSION THREAD

**Discussion by fcastrovilli** [Dokploy/dokploy#525] [Q&A] [answered]
Hi there,

I'm trying out Dokploy and already loving it! I need to manage a cluster with 3 local nodes (1 master, 2 workers). Everything is working as expected, but I need more details about the self-hosted registry setup. When this feature is enabled, I notice that a new container is created (confirmed using the `docker ps` command), and the correct port (`5000`) is exposed. However, no Traefik entrypoint is added to allow external access. Am I supposed to set this up manually? Why is there a `Registry URL` field if no Traefik proxy is linked to this container? Would also be very nice to have a local network only access to the registry, I mean just for the cluster nodes to access it with something like `192.168.1.xxx:5000` in the registry url field.

I know this might be a beginner's question, but given the "We set this up for you" statement in the self-hosted registry documentation, I was hoping for more information on the setup process.

**Comment by sudhanshug16** [comment]
Wondering the same. No SSL was generated for the domain I provided and I just get a 404 if I try to log in via docker to the registry domain.

Exact log:

```
> docker login -u root -p <> http://registry.<>.dev
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
Error response from daemon: Get "https://registry.<>.dev/v2/": tls: failed to verify certificate: x509: certificate is valid for 92137e70c3aff3b7e8a423c65aa6f779f28.1060aeb8e9006c3b5b8bed4fe869d31237d.traefik.default, not registry.<>.dev
```

**Comment by Siumauricio** [comment, marked as answer]
Most likely we will deprecate the self-hosted registry, it gives some problems in terms of domain links and so on, at the end of the day it is a docker image that can be used as a normal application and link domains to it, just like any other application.

**Reply by kinton** [reply]
Such a registry can provide zero downtime easier for some users than a non non-one-click solution.

**Reply by TorbenHammes** [reply]
> Most likely we will deprecate the self-hosted registry, it gives some problems in terms of domain links and so on, at the end of the day it is a docker image that can be used as a normal application and link domains to it, just like any other application.

Then you should remove it from your docs ;-)

**Reply by Frits1980** [reply]
> Most likely we will deprecate the self-hosted registry, it gives some problems in terms of domain links and so on, at the end of the day it is a docker image that can be used as a normal application and link domains to it, just like any other application.

Please do what @TorbenHammes already suggested. And do that for a lot of stuff. Started working with Dokploy about a week ago, but found out that the documentation is really lacking up-to-date info.
